clang/lib/CodeGen/CGDecl.cpp

```diff
@@ -1412,6 +1420,39 @@ void CodeGenFunction::EmitAndRegisterVariableArrayDimensions(
   }
 }
 
+/// Return the maximum size of an aggregate for which we generate a fake use
+/// intrinsic when -fextend-lifetimes is in effect.
+static uint64_t maxFakeUseAggregateSize(const ASTContext &C) {
+  return 4 * C.getTypeSize(C.UnsignedIntTy);
```

Comment:
This limit should be documented in the associated release note.

Reply:
Good call - the release note also ought to be part of this patch imo.